src/Toolkit/Collection.php

```diff
@@ -238,50 +238,63 @@ public function extend($items)
      * Filters elements by a custom
      * filter function or an array of filters
      *
+     * @deprecated 3.5.0 Use `Kirby\Toolkit\Collection::filterBy` instead
+     *
      * @param Closure $filter
      * @return self
      */
     public function filter($filter)
     {
-        if (is_callable($filter) === true) {
-            $collection = clone $this;
-            $collection->data = array_filter($this->data, $filter);
-
-            return $collection;
-        } elseif (is_array($filter) === true) {
-            $collection = $this;
-
-            foreach ($filter as $arguments) {
-                $collection = $collection->filterBy(...$arguments);
-            }
-
-            return $collection;
-        }
-
-        throw new Exception('The filter method needs either an array of filterBy rules or a closure function to be passed as parameter.');
+        return $this->filterBy($filter);
```

Comment:
Please add a code comment above this line that the deprecation warning was not added on purpose for now and that it should be added in one of the next minor releases. Otherwise we might forget about it and will need to search for the discussion in this PR to find out what the plan was.

Reply:
I don't think we should not fully deprecate this method in v3 at all and keep it as shorthand/alias.